### ISSUE TITLE
[Footer]: QA: Extra tab focus is observed on 'Seletc geographic' modal

### ISSUE THREAD

**Issue by Gopi916** [carbon-design-system/carbon-for-ibm-dotcom#8851]
### Description

Page Element:
Select an area in 'Select geographic' modal >> geographic area modal got triggered

Issue:
Extra tab focus is observed on hitting Tab/Shift+tab from Search filed.

Expected:
Expect to remove Extra tab focus 

### Component(s) impacted

Footer [Default, Default with Adjacent legal links, Short, Short with adjunct legal links, Micro]

### Browser

Firefox

### Carbon for IBM.com version

v1.34.0-rc.0

### Severity

Severity 4 = The problem is not visible to or noticeable to an average user. Affects minor functionality, no workaround needed.

### Application/website

Carbon for IBM.com

### Package

@carbon/ibmdotcom-web-components

### CodeSandbox example

https://carbon-design-system.github.io/carbon-for-ibm-dotcom/next/web-components/?path=/story/components-masthead--default

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/main/.github/CODE_OF_CONDUCT.md)
- [ ] I checked the [current issues](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues) for duplicate issues
